### PR TITLE
[5.1] Clear column aliases in count queries

### PR DESCRIPTION
**Pull request by rkgrep**
Discussion in #9993

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -2056,4 +2056,24 @@ public function __call($method, $parameters)
...
+    /**
+     * Clear column aliases.
+     *
+     * @param array $columns
+     * @return array
+     */
+    protected function clearSelectAliases(array $columns)
+    {
+        foreach ($columns as &$column) {
+            if (is_string($column)) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]:
```php
if (! is_string($column) || ($aliasPosition = strpos(strtolower($column), ' as ')) === false) {
    continue;
}

$column = substr($column, 0, $aliasPosition);
```

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: that would reduce the nesting here ^^^

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -2056,4 +2056,24 @@ public function __call($method, $parameters)
 
         throw new BadMethodCallException("Call to undefined method {$className}::{$method}()");
     }
+
+    /**
+     * Clear column aliases.
+     *
+     * @param array $columns
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: cs

**Comment by kohenkatz** [comment]
:+1:

**Pull request merged**